qa/evil-tests/src/test/java/org/elasticsearch/bootstrap/BootstrapCliParserTests.java

```
@@ -235,6 +246,7 @@ public void testThatHelpfulErrorMessageIsGivenWhenParametersAreOutOfOrder() thro
                 parser.parse("start", new String[]{"--foo=bar", "-Dbaz=qux"});
         });
         assertThat(e.getMessage(), containsString("must be before any parameters starting with --"));
+        assertNull(System.getProperty("es.foo"));
```

it's the one that can't leak here I think fail fast is good?